wasmbinding/message_plugin.go

```diff
@@ -295,6 +294,34 @@ func (m *CustomMessenger) submitTx(ctx sdk.Context, contractAddr sdk.AccAddress,
 }
 
 func (m *CustomMessenger) submitAdminProposal(ctx sdk.Context, contractAddr sdk.AccAddress, submitAdminProposal *bindings.SubmitAdminProposal) ([]sdk.Event, [][]byte, error) {
+	var data []byte
+	if submitAdminProposal.AdminProposal.ParamChangeProposal != nil || submitAdminProposal.AdminProposal.UpgradeProposal != nil || submitAdminProposal.AdminProposal.ClientUpdateProposal != nil {
```

Comment:
Please add a comment there why exactly we need that condition.

Reply:
Also i think it's cleaner to create a method `isLegacyProposal()` which does this check inside of it

Reply:
These two conditions differ only in the method that should called (`m.performSubmitAdminProposalLegacy()` or `m.performSubmitAdminProposal()`). 

Logging and unmarshalling logic is the same for both conditions. Let's refactor to make it easier to read?